core/controllers/moderator.py

```diff
@@ -78,15 +68,26 @@ def post(self) -> None:
             'featured_activity_reference_dicts']
 
         try:
-            summary_services.require_activities_to_be_public(
+            # Get the list of invalid activity IDs (non-existent or private).
+            invalid_ids = summary_services.require_activities_to_be_public(
                 featured_activity_references)
+
+            # If there are invalid IDs, raise an exception with those IDs.
+            if invalid_ids:
+                error_message = (
+                    'Exploration IDs: ' +
+                    ', '.join(invalid_ids) + ' are invalid.'
```

Comment:
Can we have a more specific error message for explorations that are valid, but are private?